runtime/commonMain/src/kotlinx/serialization/internal/Tuples.kt

```diff
@@ -94,7 +102,8 @@ public class PairSerializer<K, V>(
 }
 
 
-// todo: move from internal package and add documentation
+@InternalSerializationApi
+@Deprecated(message = deprecationMessage, level = DeprecationLevel.HIDDEN)
```

Comment:
It will, and it will be fixed with `alt + enter -> import`.
Unfortunately, it is not possible to deprecate it with `ERROR`: `LowPriorityInOverloadResolution` is inapplicable for classes and constructors, so deprecated symbol always wins in resolution if you have two import.

Reply:
I can see that this is an 'unresolved reference' until you explicitly have `import kotlinx.serialization.internal.TripleSerializer` (e.g. alt+enter would not work if one has a star import of internal)

Reply:
If we move out everything from `internal`, users won't need that star import anymore; therefore, it's nothing wrong in overload resolution